tools/src/main/java/com/nedap/archie/rmobjectvalidator/RMObjectValidator.java

```diff
@@ -130,6 +148,34 @@ else if (cobject instanceof CPrimitiveObject) {
         return result;
     }
 
+    private void validateArchetypeSlot(List<RMObjectWithPath> rmObjects, String path, CObject cobject, List<RMObjectValidationMessage> result) {
+        ArchetypeSlot slot = (ArchetypeSlot) cobject;
+        for (RMObjectWithPath objectWithPath : rmObjects) {
+
+            Object object = objectWithPath.getObject();
+
+            String archetypeId =  metaModel.getSelectedModel().getArchetypeIdFromArchetypedRmObject(object);
+            if(archetypeId != null) {
+                if(!AOMUtils.archetypeRefMatchesSlotExpression(archetypeId, slot)) {
+                    //invalid archetype id, add message
+                    this.addMessage(slot, objectWithPath.getPath(), RMObjectValidationMessageIds.rm_ARCHETYPE_ID_SLOT_MISMATCH.getMessage(archetypeId), RMObjectValidationMessageType.WRONG_ARCHETYPE_ID);
+                }
+                //but do continue validation!
+                OperationalTemplate operationalTemplate = operationalTemplateProvider.getOperationalTemplate(archetypeId);
+                if(operationalTemplate != null) {
+                    //occurrences already validated, so nothing left to validate from the archetyepe root
+                    //from now on, validate from the root of the found OPT
+                    CObject newRoot = operationalTemplate.getDefinition();
+                    validateObjectWithPath(result, newRoot, path, objectWithPath);
```

Comment:
It can be fixed. But I am not sure if we should, since there is no actual archetype path involved here that can be directly looked up. Maybe add the archetype id in the message? then at least iets clear where this message is coming from?

Reply:
Added the archetype id, plus documentation in javadoc.